microsoft/testsuites/ltp/ltp.py

```diff
@@ -0,0 +1,355 @@
+    LTP_GIT_URL = "https://github.com/linux-test-project/ltp.git"
+    BUILD_REQUIRED_DISK_SIZE_IN_GB = 2
+    LTP_RESULT_PATH = "/opt/ltp/ltp-results.log"
+    LTP_OUTPUT_PATH = "/opt/ltp/ltp-output.log"
+    LTP_SKIP_FILE = "/opt/ltp/skipfile"
+
+    @property
+    def command(self) -> str:
+        return "/opt/ltp/runltp"
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Make, Gcc, Git]
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def run_test(
+        self,
+        ltp_tests: List[str],
+        skip_tests: List[str],
+        log_path: str,
+        drive_name: Optional[str] = None,
+    ) -> List[LtpResult]:
+        # tests cannot be empty
+        assert_that(ltp_tests, "ltp_tests cannot be empty").is_not_empty()
+        ls = self.node.tools[Ls]
+        rm = self.node.tools[Rm]
+
+        # remove skipfile if it exists
+        if ls.path_exists(self.LTP_SKIP_FILE):
+            self._log.debug(f"Removing skipfile: {self.LTP_SKIP_FILE}")
+            rm.remove_file(self.LTP_SKIP_FILE, sudo=True)
+
+        # remove results file if it exists
+        if ls.path_exists(self.LTP_RESULT_PATH, sudo=True):
+            self._log.debug(f"Removing {self.LTP_RESULT_PATH}")
+            rm.remove_file(self.LTP_RESULT_PATH, sudo=True)
+
+        # remove output file if it exists
+        if ls.path_exists(self.LTP_OUTPUT_PATH, sudo=True):
+            self._log.debug(f"Removing {self.LTP_OUTPUT_PATH}")
+            rm.remove_file(self.LTP_OUTPUT_PATH, sudo=True)
+
+        # add parameters for the test logging
+        parameters = f"-p -q -l {self.LTP_RESULT_PATH} -o {self.LTP_OUTPUT_PATH} "
+
+        # add the list of tests to run
+        parameters += f"-f {','.join(ltp_tests)} "
+
+        # add logging and output file parameter
+        if drive_name:
+            parameters += f"-z {drive_name} "
+
+        # add the list of skip tests to run
+        if len(skip_tests) > 0:
+            # write skip test to skipfile with newline separator
+            skip_file_value = "\n".join(skip_tests)
+            self.node.tools[Echo].write_to_file(
+                skip_file_value, PurePosixPath(self.LTP_SKIP_FILE), sudo=True
+            )
+            parameters += f"-S {self.LTP_SKIP_FILE} "
+
+        # Minimum 4M swap space is needed by some mmp test
+        if self.node.tools[Free].get_swap_size() < 4:
+            self.node.tools[Swap].create_swap()
+
+        # run ltp tests
+        self.run(parameters, sudo=True, force_run=True, timeout=12000)
+
+        # write output to working_path
+        cat = self.node.tools[Cat]
+        with open(f"{log_path}/ltp-output.log", "w") as f:
+            f.write(
+                cat.read(
+                    self.LTP_OUTPUT_PATH, sudo=True, force_run=True, no_debug_log=True
+                )
+            )
+
+        # write results to working_path
+        with open(f"{log_path}/ltp-results.log", "w") as f:
```

Comment:
the working_path is different with log_path. The working_path put temp files, which are not considered as log, it might be used for troubleshooting. But the log_path might be copied to some places to save. So the log path should put log only, and make it's a reasonable size.

Reply:
I want to copy the results and output so that they can be saved as additional logs for future reference.

Reply:
I will update the comment if it's confusing

Reply:
use node.os.shell.copy_back. It can copy a file back to lisa.

Reply:
fixed. Much faster than using `cat`